src/app/cli/src/coda_worker_testnet.ml

```diff
@@ -55,29 +55,27 @@ module Api = struct
     Coda_process.disconnect (List.nth_exn t.workers i)
 
   let send_payment t i sender_sk receiver_pk amount fee =
+    let open Deferred.Option.Let_syntax in
+    let worker = List.nth_exn t.workers i in
     let sender_pk =
       Public_key.of_private_key_exn sender_sk |> Public_key.compress
     in
-    (let open Deferred.Option.Let_syntax in
-    let%bind maybe_nonce = get_nonce t i sender_pk in
-    let nonce = Option.value_exn maybe_nonce in
+    let%bind nonce = Coda_process.get_nonce_exn worker sender_pk in
     let payload =
       User_command.Payload.create ~fee ~nonce ~memo:User_command_memo.dummy
         ~body:(Payment {receiver= receiver_pk; amount})
     in
-    let cmd =
-      User_command.sign (Keypair.of_private_key_exn sender_sk) payload
+    let user_cmd =
+      ( User_command.sign (Keypair.of_private_key_exn sender_sk) payload
+        :> User_command.t )
     in
-    let%map _ =
-      run_online_worker
-        ~arg:(cmd :> User_command.t)
-        ~f:(fun ~worker cmd -> Coda_process.process_payment_exn worker cmd)
-        t i
+    let%map receipt =
+      Coda_process.process_payment_exn worker user_cmd
+      |> Deferred.map ~f:Or_error.ok
     in
-    Linear_pipe.write t.payment_writer (cmd :> User_command.t)
-    |> don't_wait_for)
-    |> ignore ;
-    Deferred.unit
+    let result = Ivar.create () in
```

Comment:
`result` is not a descriptive name. This ivar basically signals whoever waits for it that all nodes received the same txn at their transition_frontier root within a certain amount of time. Can you name it something like that?